docs/kb/laptop-dev-env.md

```diff
@@ -0,0 +1,224 @@
+# NovaCustom laptop development environment
+
+This document describes the process of configuring a development and debugging
+environment for NovaCustom laptops.
+
+The instruction is applicable for workstations running Fedora. If you're using
+a different distro, some commands (e.g. package installation) may be different.
+
+## Embedded Controller
+
+This section describes the development environment for EC firmware on NovaCustom
+devices. The Embedded Controller firmware is heavily based on System76 EC
+firmware and makes use of open-source debugging tools and software also
+developed by System76.
+
+### Flashing
+
+This section outlines the parts and software needed to perform external flashing
+of the Embedded Controller.
+
+| Size | 14 inch models | 16 inch models |
+| --- | --- | --- |
+| Programmer | Arduino Mega 2560 | Arduino Mega 2560 |
+| Cable | **FFC 0.5mm pitch 24 pin** | **FFC 1.0mm pitch 24 pin** |
+| Software | ecflash | ecflash |
+
+#### Preparation
+
+1. Obtain a copy of the EC firmware and tools:
+
+    ```bash
+    git clone https://github.com/Dasharo/ec.git
+    cd ec
+    git submodule update --init --checkout
+    ./scripts/deps.sh
+    ```
+
+1. Follow the on-screen instructions to finish configuring your development
+environment.
+1. Connect the Arduino to your workstation with a USB-A to USB-B cable. Do not
+connect the target laptop yet.
+1. Flash the Arduino with debugger firmware:
+
+    ```bash
+    make BOARD=arduino/mega2560
+    make BOARD=arduino/mega2560 flash
+    ```
+
+#### Flash the target laptop
+
+1. In the EC directory, build the EC flashing application:
```

Comment:
I thing this goes to preparation

Reply:
Right, moved: https://github.com/Dasharo/docs/compare/4e9ec5622c840b8a6d840a400efafdd5de3ee80f..6a19fdc3b9c3ed99815a29ce364e8e90543a5ac2